backend/src/main/java/com/htecgroup/skynest/service/impl/CurrentUserServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.entity.UserEntity;
+import com.htecgroup.skynest.repository.UserRepository;
+import com.htecgroup.skynest.service.CurrentUserService;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class CurrentUserServiceImpl implements CurrentUserService {
+
+  private UserRepository userRepository;
+
+  @Override
+  public Authentication getAuthentication() {
+    return SecurityContextHolder.getContext().getAuthentication();
+  }
+
+  @Override
+  public String getEmail() {
+    return getAuthentication().getPrincipal().toString();
+  }
+
+  @Override
+  public String getRole() {
+    return new ArrayList<>(getAuthentication().getAuthorities()).get(0).getAuthority();
```

Comment:
What if the user has more than one Role? Maybe you can change this method to return List instead

Reply:
Sure, will do it